NDTensors/src/abstractarray/data_isa.jl

```diff
@@ -0,0 +1,4 @@
+data_isa(t::Tensor, datatype::Type) = data_isa(storage(t), datatype)
+data_isa(s::TensorStorage, datatype::Type) = data_isa(data(s), datatype)
+data_isa(d::AbstractArray, datatype::Type) = d isa datatype # Might have to unwrap if it is reshaped, sliced, etc.
+data_isa(n::Number, datatype::Type) = false
```

Comment:
What's this definition for?

Reply:
Seems a bit strange to ask for the data type of a `Number`, since it doesn't have data.

Reply:
As discussed this is due to diag. I added a comment